### DISCUSSION TITLE
Specifying different package sets for packages within the same project

### DISCUSSION THREAD

**Discussion by newton-migosi** [srid/haskell-flake#205] [Q&A] [answered]
I have  a project with the following directory structure
```
my-app/
├─ frontend/
│  ├─ app/
│  │  ├─ Main.hs
│  ├─ frontend.cabal
├─ backend/
│  ├─ app/
│  │  ├─ Main.hs
│  ├─ backend.cabal
flake.nix
```

I'd like to build the subdirs `frontend` and `backend` using different package sets. How do I do this?

**Comment by newton-migosi** [comment]
The current version in the example only caters to using the default project.cabal https://github.com/srid/haskell-multi-nix/blob/c85563721c388629fa9e538a1d97274861bc8321/flake.nix

**Reply by newton-migosi** [reply]
It looks like the documentation is out of sync, comparing 
https://zero-to-flakes.com/haskell-flake/modules/
```nix
{
  # Inside flake-parts' `perSystem`:
  haskellProjects.default = {
    imports = [
      inputs.common.haskellFlakeProjectModules.default
    ];
    packages = {
      myapp.root = ./.;
    };
  };
}
```

and
https://zero-to-flakes.com/haskell-flake/package-set
```nix
{
  haskellProjects.myproject = {
    packages.mypackage.source = ./.;

    basePackages = config.haskellProjects.ghc810.outputs.finalPackages;
  };
}
```

**Comment by srid** [comment, marked as answer]
> The current version in the example only caters to using the default project.cabal https://github.com/srid/haskell-multi-nix/blob/c85563721c388629fa9e538a1d97274861bc8321/flake.nix

Local packages can be specified in the same manner as packages pointing to custom source paths: https://zero-to-flakes.com/haskell-flake/dependency#path

If you want them to use different package sets, you want to define distinct projects (because each project has its own package set and overrides):

```nix
haskellProjects.frontend-project = {
  packages = {
    frontend.source = ./frontend;
  };
};

haskellProjects.backend-project = {
  packages = {
    backend.source = ./backend;
  };
};
```

**Reply by newton-migosi** [reply]
thanks.. the library is changing pretty often (active development is good), I had missed out on the changes
